### ISSUE TITLE
Headerbar on Wayland not resizing / hiding

### ISSUE THREAD

**Issue by ab0v3g4me** [ggez/ggez#393]
If i were to compile this it would show the headerbar with the sizes of  `ggez::conf::WindowMode`, that were initially set.Here they are initially set to `(800, 600)` ( according to the docs ), even if i override them by creating a `WindowMode` manually it would stick the the dimensions i supplied.

```rust
fn main() {
        let mut context = match Context::load_from_conf("MyGame", "Abovegame", config ).unwrap();
        graphics::set_fullscreen(&mut context, true);
       /* . . . . */
       event::run(&mut context, &mut MyStruct);
}
```
- I'm on Archlinux and using <b>Wayland</b> as my display server.
![screenshot from 2018-06-07 20-14-18](https://user-images.githubusercontent.com/19503519/41120475-e93fbf00-6a95-11e8-9cc0-bc791cbcad49.png)
I've managed to overcome this with a hacky solution, but i guess this is not how it was supposed to be done : ).
```rust
fn main() {
        let (w, h) =  {
            let mut context = Context::load_from_conf("MyGame", "Abovegame", conf::Conf::new() ).unwrap();
            graphics::set_fullscreen(&mut context, true);
            graphics::get_size(&context)
        };

        let mut config = conf::Conf::new();
        config.window_mode.width = w;
        config.window_mode.height = h;

        let mut context = Context::load_from_conf("MyGame", "Abovegame", config ).unwrap();
        graphics::set_fullscreen(&mut context, true);
        let mut menu = main_menu::MainMenu::with_ui(ui::UIButton::init(w,h));

        event::run(&mut context, &mut menu);
}
```
Basically spawning a `Context` in a local scope and getting it's size in fullscreen mode then use that size with the latter code.

**Comment by icefoxen** [comment]
Thank you!  Wayland is something I should really test more thoroughly... or, uh, at all.

**Comment by ab0v3g4me** [comment]
None that i can see.

![screenshot from 2018-06-08 14-36-04](https://user-images.githubusercontent.com/19503519/41158117-1bc2393e-6b28-11e8-9c07-61954e91b046.png)

![screenshot from 2018-06-08 14-37-36](https://user-images.githubusercontent.com/19503519/41158150-39a56caa-6b28-11e8-8aca-29f30d2d266b.png)

**Comment by bzm3r** [comment]
Why is this a bug on the `ggez` side?

**Comment by icefoxen** [comment]
@bzm3r It probably isn't but it's nice to know for sure before forwarding it upstream.

**Comment by icefoxen** [comment]
Might be fixed by https://github.com/ggez/ggez/commit/0cb700d70fcfe75b2d9aab392213e0f2b6a40497 , please double-check and re-open if this is still a problem.
